cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeconfig
+
+import (
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"path"
+
+	certphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/certs"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	certutil "k8s.io/kubernetes/pkg/util/cert"
+)
+
+const KubernetesDirPermissions = 0700
+
+// This function is called from the main init and does the work for the default phase behaviour
+func CreateAdminAndKubeletKubeConfig(masterEndpoint, pkiDir, outDir string) error {
+	// Parse the certificate from a file
+	caCertPath := path.Join(pkiDir, "ca.pem")
+	caCerts, err := certutil.CertsFromFile(caCertPath)
+	if err != nil || len(caCerts) == 0 {
+		return fmt.Errorf("couldn't read or parse the ca cert file %s: %v", caCertPath, err)
```

Comment:
at a closer look, err will be non-nil if `len(caCerts) == 0` so that check didn't make sense